activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -227,24 +233,40 @@ def before_commit_records
       end
 
       def commit_records
```

Comment:
Given `rollback_records` and `before_commit_records` exist, I imagine corresponding use cases are going to call for matching txn-level callbacks... but I think it's fine to wait and explore that when it comes up.